src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -14,50 +15,97 @@ namespace Microsoft.Extensions.Caching.Memory
     internal sealed partial class CacheEntry : ICacheEntry
     {
         private static readonly Action<object> ExpirationCallback = ExpirationTokensExpired;
+        private static readonly AsyncLocal<CacheEntry?> _current = new AsyncLocal<CacheEntry?>();
 
         private readonly MemoryCache _cache;
 
         private CacheEntryTokens? _tokens; // might be null if user is not using the tokens or callbacks
-        private TimeSpan? _absoluteExpirationRelativeToNow;
-        private TimeSpan? _slidingExpiration;
-        private long? _size;
+        private TimeSpan _absoluteExpirationRelativeToNow;
+        private TimeSpan _slidingExpiration;
+        private long _size = -1;
         private CacheEntry? _previous; // this field is not null only before the entry is added to the cache and tracking is enabled
         private object? _value;
-        private CacheEntryState _state;
+        private DateTime _absoluteExpiration;
+        private short _absoluteExpirationOffsetMinutes;
+        private bool _isDisposed;
```

Comment:
I don't know how many bytes you want to save, but all these bools could be stored as bits.

Reply:
Currently this wouldn't save anything because of padding.